src/lib/draw-and-pick.js

```diff
@@ -136,16 +136,21 @@ export function pickLayers(gl, {
 
     affectedLayers.forEach(layer => {
       let info = Object.assign({}, baseInfo);
-      info.layer = layer;
 
       if (layer === pickedLayer) {
         info.color = pickedColor;
         info.index = pickedObjectIndex;
         info.picked = true;
       }
 
-      // Let layers populate its own info object
-      info = layer.pickLayer({info, mode});
+      // walk up the composite chain and find the owner of the event
+      // sublayers are never directly exposed to the user
+      while (layer && info) {
+        info.layer = layer;
+        // Let layers populate its own info object
+        info = layer.pickLayer({info, mode});
```

Comment:
We absolutely need chaining, the question is more whether these should all modify in the same picking object or we should create a chain of picking objects, one for the parent and one for each sublayer. Imagine:

`info.parent`


I can't see many applications using that chain though, so it is probably overkill. But it would avoid the potential mess of multiple layers overwriting (or not overwriting) each other's fields in the same info object.